sdk/core/azure-core/inc/azure/core/url.hpp

```diff
@@ -239,5 +239,67 @@ namespace Azure { namespace Core {
      * @return Absolute URL with URL-encoded query parameters.
      */
     std::string GetAbsoluteUrl() const;
+
+    /**
+     * @brief Compares with another instance of Url for equality.
+     * @param other another instance of Url.
+     * @return `true` if values of two Url instances are equal, `false` otherwise.
+     *
+     * @note Two instances are considered equal if their string representation would match.
+     *
+     */
+    // See also https://datatracker.ietf.org/doc/html/rfc3986#section-6.2.1
```

Comment:
This comment doesn't hold (and from my understanding of the RFC, it shouldn't).

A naiive ToString()/GetAbsoluteUrl() comparison is likely not sufficient. Two url’s with different fragments [are considered equal in .NET’s == operator](https://learn.microsoft.com/en-us/dotnet/api/system.uri.equals?view=net-8.0#system-uri-equals(system-object)). That follows the RFC. Our C++ Url, in the current PR, doesn’t.

> "fragment components (if any) should be excluded from the comparison."

Reply:
What do you mean it doesn't hold?

Our operator == provides (effectively) a character-by-character comparison as described in Section 6.2.1 (we just happen to do that without actually constructing a string).

6.2.1 is a valid form of comparison, we implement it.

Reply:
From 6.1:

> Because URIs exist to identify resources, presumably they should be
   considered equivalent when they identify the same resource.  However,
   this definition of equivalence is not of much practical use, as there
   is no way for an implementation to compare two resources unless it
   has full knowledge or control of them.  For this reason,
   determination of equivalence or difference of URIs is based on string
   comparison, perhaps augmented by reference to additional rules
   provided by URI scheme definitions.  We use the terms "different" and
   "equivalent" to describe the possible outcomes of such comparisons,
   but there are many application-dependent versions of equivalence.